documents/process/release.md

```diff
@@ -28,35 +28,31 @@ This is a checklist of things that should be done in the weeks leading to the re
 * [ ] Verify that `ffi/capi` depends on a released (not Git) version of Diplomat. Get it published (ask manishearth or sffc) otherwise.
-* [ ] Start work on a changelog (see below)
+* [ ] Get all contributors to complete the changelog (see below)
 * [ ] Draft the text for the GitHub release. This text will be sent to GitHub subscribers and can also be used for the mailing list email.
 * [ ] Prepare a PR to update tutorials using the upcoming release. The PR should pass `cargo make test-tutorials-local`, but can fail `cargo make test-tutorials-cratesio` prior to release
 
 ## Release steps
 
 Once the release checklist is complete, the assigned release driver will perform the following steps, in order:
 
+* Land the changelog (see below)
+  * Note: Do this _before_ tagging the release so that it is included in the tag
 * Go through the prerelease checklist again, ensuring that no problems were reintroduced in the PRs that landed since the opening of the checklist. (Things like doc prettification will likely need to be rerun!)
 * Release utils (see section below). Get the version bumps reviewed and checked in before releasing.
 * Update ICU4X package versions as needed. Most of this can be done by updating `workspace.package` in the root `Cargo.toml` and the `workspace.dependencies` entries there. Some `icu_*` crates do not follow the ICU4X versioning scheme like `icu_codepointtrie_builder` or experimental crates.
-* Temporarily add `icu_testdata` back to the workspace
-  * To build, run `cargo make testdata-legacy-gen` to generate the gitignored data
 * Get this reviewed and checked in before continuing
 * Publish each `icu_*` crate
   * Use `cargo workspaces publish --from-git` to automatically publish the crates in the correct order if you would like
   * Add `icu4x-release` group as owners to each new component you're publishing
     * `cargo owner -a github:unicode-org:icu4x-release`
 * Merge the tutorials PR. `cargo make test-tuturials-cratesio` should now pass
   * If there are any errors, please fix them before advertising the release
-* Land the changelog (see below)
-  * Note: Do this _before_ tagging the release because the changelog is included in the tag
 * [Tag the Release](https://github.com/unicode-org/icu4x/releases) with the text drafted above
 * Create a branch named `release/x.y` on the release tag and push it to the upstream
 * Announce the release to public
   * Send an email to [icu4x-announce](https://groups.google.com/u/0/a/unicode.org/g/icu4x-announce)
   * Submit to This Week In Rust
-* Keep the main branch relatively stable for 7-14 days following the release to make things easier in case a patch release is needed.
-  * It's okay to land smaller or incremental changes, but avoid breaking changes during this period.
```

Comment:
Issue: I assume you deleted this from the guide since we now have our branching approach, but it's still the case that if we need to fix something after a release, we want to make it easy to patch on both main and on the release branch.

Reply:
But I don't want to wait another 14 days to make breaking changes :)